### ISSUE TITLE
During compile i constantly get this error xtensa-esp32-elf-g++: not found

### ISSUE THREAD

**Issue by wesselah** [benzino77/tasmocompiler#376]
I just upgraded the docker container and when trying to compile i constantly get this error

Processing tasmota32 (framework: arduino; platform: https://github.com/tasmota/platform-espressif32/releases/download/2023.07.00/platform-espressif32.zip; board: esp32)
--------------------------------------------------------------------------------
Tool Manager: Installing espressif/toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch5
Downloading 0% 10% 20% 30% 40% 50% 60% 70% 80% 90% 100%
Unpacking 0% 10% 20% 30% 40% 50% 60% 70% 80% 90% 100%
Tool Manager: toolchain-xtensa-esp32@8.4.0+2021r2-patch5 has been installed!
Verbose mode can be enabled via `-v, --verbose` option
-DCONFIG_TASMOTA_FLASHMODE_DIO
*** use provided user_config_override.h as planned ***
*** use provided platformio_override.ini as planned ***
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/esp32.html
PLATFORM: Espressif 32 (2023.7.0) > Espressif Generic ESP32 >= 4M Flash PSRAM, Tasmota 2880k Code/OTA, 320k FS
HARDWARE: ESP32 160MHz, 320KB RAM, 4MB Flash
DEBUG: Current (cmsis-dap) External (cmsis-dap, esp-bridge, esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES: 
 - framework-arduinoespressif32 @ 2.0.11 
 - tool-esptoolpy @ 1.40602.0 (4.6.2) 
 - tool-mklittlefs @ 1.203.210628 (2.3) 
 - tool-mkspiffs @ 2.230.0 (2.30) 
 - toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch5
Converting tasmota.ino
sh: 1: xtensa-esp32-elf-g++: not found
*** Error 127
AssertionError: :
  File "/usr/local/lib/python3.7/dist-packages/platformio/builder/main.py", line 173:
    env.SConscript("$BUILD_SCRIPT")
  File "/root/.platformio/packages/tool-scons/scons-local-4.5.2/SCons/Script/SConscript.py", line 598:
    return _SConscript(self.fs, *files, **subst_kw)
  File "/root/.platformio/packages/tool-scons/scons-local-4.5.2/SCons/Script/SConscript.py", line 285:
    exec(compile(scriptdata, scriptname, 'exec'), call_stack[-1].globals)
  File "/root/.platformio/platforms/espressif32/builder/main.py", line 324:
    target_elf = env.BuildProgram()
  File "/root/.platformio/packages/tool-scons/scons-local-4.5.2/SCons/Util/envs.py", line 242:
    return self.method(*nargs, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/platformio/builder/tools/piobuild.py", line 61:
    env.ProcessProgramDeps()
  File "/root/.platformio/packages/tool-scons/scons-local-4.5.2/SCons/Util/envs.py", line 242:
    return self.method(*nargs, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/platformio/builder/tools/piobuild.py", line 121:
    env.BuildFrameworks(env.get("PIOFRAMEWORK"))
  File "/root/.platformio/packages/tool-scons/scons-local-4.5.2/SCons/Util/envs.py", line 242:
    return self.method(*nargs, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/platformio/builder/tools/piobuild.py", line 339:
    env.ConvertInoToCpp()
  File "/root/.platformio/packages/tool-scons/scons-local-4.5.2/SCons/Util/envs.py", line 242:
    return self.method(*nargs, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/platformio/builder/tools/pioino.py", line 237:
    out_file = c.convert(ino_nodes)
  File "/usr/local/lib/python3.7/dist-packages/platformio/builder/tools/pioino.py", line 85:
    return self.process(contents)
  File "/usr/local/lib/python3.7/dist-packages/platformio/builder/tools/pioino.py", line 106:
    assert self._gcc_preprocess(contents, out_file)
========================= [FAILED] Took 79.12 seconds =========================

Environment    Status    Duration
-------------  --------  ------------
tasmota32      FAILED    00:01:19.122
==================== 1 failed, 0 succeeded in 00:01:19.122 ====================
Finished. Exit code: 1.

Any help would be apprciated

**Comment by benzino77** [comment]
Hi, 
Sorry for the lag in answering your question. Have a look here: 
#270 
#256 
#231 
#303 
or [issues](https://github.com/benzino77/tasmocompiler/issues?q=xtensa)

It is probably a problem with espressif availability of a toolchain for a specific architecture.

**Comment by wesselah** [comment]
I removed the running docker instance and installed the latest version of tasmocompiler .
It is running on raspberry pi 4 32 bit not on 64 bit and the somewhat older tasmocompiler  docker run fine so i thought lets update it to the latest version.
I downloaded also the latest source for tasmota and then i followed up to the next steps even the basic esp 32 selection gets the same error. I was looking for the missing files but it seems not to be downloaded correctly i tried over again and still the issue is there. Then i tried gitpod and it works normally. So it seems like a local install issue with the latest docker image. Could this be ? It is only a couple of days old
